src/main/java/seedu/address/model/ScheduleManager.java

```diff
@@ -62,11 +62,29 @@ public TimeIntervalList getAvailableTimeSlots() {
 
     /**
      * Checks whether the current time is within operation hours.
+     *
      * @return true if current time is within operation hours, false otherwise.
      */
-    public boolean isWithinOperationHour() {
-        LocalTime currentTime = LocalTime.now();
-        return (currentTime.getHour() >= 8 && currentTime.getHour() < 22);
+    public boolean isWithinOperationHour(LocalTime time) {
+        return !isBeforeOperationHour(time) && !isAfterOperationHour(time);
+    }
```

Comment:
Should the JavaDocs be updated too?

Reply:
Oh right thanks for pointing that out! 👍